inference-engine/src/mkldnn_plugin/mkldnn_graph_optimizer.cpp

```diff
@@ -895,13 +898,42 @@ void MKLDNNGraphOptimizer::FuseConvolutionAndDWConvolution(MKLDNNGraph &graph) {
                                  everyone_is(1, allPads.end[X_AXIS], allPads.end[Y_AXIS]) &&
                                  everyone_is(1, childLayer->_dilation[X_AXIS], childLayer->_dilation[Y_AXIS]) &&
                                  childLayer->_stride[X_AXIS] == childLayer->_stride[Y_AXIS] &&
-                                 false &&  // TODO [oneDNN]: disabled while not ported
-                                 one_of(childLayer->_stride[X_AXIS], 1 /*, 2*/) &&  // TODO [oneDNN]: stride 2 should also be supported
+                                 withBias &&
+                                 one_of(childLayer->_stride[X_AXIS], 1, 2) &&
                                  childNode->getChildEdgeAt(0)->getDims().ndims() == 4;
 
         return isSupportedParams;
     };
 
+    auto isFusingWorthwhile = [&](MKLDNNNodePtr parentNode, MKLDNNNodePtr childNode) {
+        auto* layer = dynamic_cast<ConvolutionLayer*>(childNode->getCnnLayer().get());
```

Comment:
You can use std::dynamic_pointer_cast without getting the raw pointer.

Reply:
changed, thanks